### PR TITLE
Fix: (return) (command inside legacy text) (semicolon inside quoted string)

### LINKED ISSUE #40: Formatting: indentation added on unquoted v1 string with "for"
Found: Formatting this:
```ahk
InputBox, name, for bla
if somthing
    Return
```
turns to:
```ahk
InputBox, name, for bla
    if somthing
    Return
```
No change expected here.

OK. There you go :) (regarding [#58](https://github.com/AutoHotkey-Plus/vscode-autohotkey/issues/58))
Maybe the formatter should ignore the `for` if there is a keyword in the beginning of the line? 🤔

(still dunno where to look myself. sorry)

### LINKED ISSUE #56: Formatting: function returns don't indent after subroutines
It appears that a function that is defined after a subroutine has this quirk or not having proper indentation when using a return statment.

example:
```ahk

func1(x, y) {
    return x + y
}

Subroutine:
    msgbox, welcome
return

func2(x, y) {
return x * y
}
```

Each statement in these three blocks should indent but the last one does not.
However, it will *if* you remove the subroutine.

Edit:

This also occurs after hotkeys.

```ahk

func1(x, y) {
    return x + y
}

^SPACE::
    msgbox, hello
return

func2(x, y) {
return x * y
}
```

### LINKED ISSUE #119: Semi-colon in string in parentheses formats wrong
### Description

indent happens where it shouldn't
semicolon in string inside parenthesis
triggers indent on next line

### Unformatted input snippet

sql:=strrplace("asd;",";","")
return

### Expected formatted output snippet

sql:=strrplace("asd;",";","")
return

### Actual formatted output snippet

sql:=strrplace("asd;",";","")
  return

### LINKED ISSUE #204: When semi-colons are present in strings in a select case, the formatter will indent all lines following
### Description

When semi-colons are present in strings in a select case, the formatter will indent all lines following

### Unformatted input snippet

```autohotkey
Switch Choice { 
	Case "DMARC Record":
		ClipboardPaste("v=DMARC1; p=quarantine; sp=none; fo=1;")
	Case "Twilio Account SID":
		Send, abc123abc123
}
```

### Expected formatted output snippet

```autohotkey
Switch Choice { 
	Case "DMARC Record":
		ClipboardPaste("v=DMARC1; p=quarantine; sp=none; fo=1;")
	Case "Twilio Account SID":
		Send, abc123abc123
}
```

### Actual formatted output snippet

```autohotkey
	Switch Choice { 
	Case "DMARC Record":
		ClipboardPaste("v=DMARC1; p=quarantine; sp=none; fo=1;")
		Case "Twilio Account SID":
			Send, abc123abc123
		}
```

### Additional context

Remove the semi-colons in the string and formatting works as expected.

### PR DESCRIPTION
**Pull request by kyklish**
Closes #40.
Closes #56.
Closes #119.
Closes #204.

Changes proposed in this pull request:

-   change order of .replace() regexp
-   remove all characters after command keyword
-   fix typo to properly indent `return` command

Notifying @mark-wiemer

### CODE REVIEW

**Comment by mark-wiemer** [comment]
Should we maybe move the "replace semicolon" to the very end of the chain? I'm concerned with this case:

```ahk
MsgBox, { ; comment with close brace }
```

**Comment by mark-wiemer** [comment]
I'll check this out tomorrow when I have some time 🙂

**Comment by kyklish** [comment]
> Should we maybe move the "replace semicolon" to the very end of the chain? I'm concerned with this case:
> 
> ```ahk
> MsgBox, { ; comment with close brace }
> ```

You right. Moved to very end of chain.

**Comment by kyklish** [comment]
purify() may return single space character, so additionally trim purified line to make it empty (I need this for future formatter fixes).

File: src/common/codeUtil.ts [hunk 1]
```diff
@@ -8,12 +8,13 @@ export class CodeUtil {
             return '';
         }
         return original
-            .replace(/;.+/, '')
             .replace(/".*?"/g, '""') // replace string literals with empty string literal
             .replace(/{.*}/g, '') // remove matching braces
             .replace(/ +/g, ' ')
             .replace(/\bgui\b.*/gi, '')
-            .replace(/\b(msgbox)\b.+?%/gi, '$1');
+            .replace(/\b(msgbox)\b.+?%/gi, '$1')
```
Comment (mark-wiemer) [inline src/common/codeUtil.ts:15]: Let's add some unit tests for `purify` so we know what exactly we fixed and we don't break it in the future :)

**Review by mark-wiemer — CHANGES_REQUESTED** [review]

**Comment by mark-wiemer** [comment]
Yes, this is awesome! I'll have a bit of feedback once my workday is done, but I think we will be able to merge this as-is!! Thank you so much, you are very good at this :)

**Comment by mark-wiemer** [comment]
Can you provide a larger snippet? And maybe a screenshot of your editor? I think there is something simple we are missing. Have you tried the playground at TypeScriptlang.org?
________________________________

**Comment by kyklish** [comment]
Yep, thanks.

**Comment by kyklish** [comment]
Every bugfix in formatter don't pass formatter's test with file `ahk-explorer.ahk`. How to deal with this file?

I can add #160 bugfix in this PR to produce more-less relevant format result for `ahk-explorer.ahk` and commit it in this PR.

**Comment by mark-wiemer** [comment]
Update ahk-explorer.out.ahk so the test passed, then I will look at the difference and make sure we didn't introduce any bad formatting behaviors.

Same applies to any failing formatting test :)
________________________________

**Comment by kyklish** [comment]
I run `npm run eslint_inner` and got this (no errors):

`vscode-autohotkey-plus-plus@2.8.3 eslint_inner
eslint --ext ts --max-warnings=0`

**Comment by mark-wiemer** [comment]
> I can't commit updated `ahk-explorer.out.ahk` :( `npm run validate` in VS Code done without errors When I try commit I got this error:
> 
> ```
> npm run eslint_inner
> [FAILED]F:\GAMES\Android\VS Code\vscode-autohotkey-plus-plus\src\test\suite\format\samples\ahk-explorer.out.ahk
> 
> ■ npm run tsc
>       ↓1:2 package.json   — no fileserror [SKIPPED]  Parsing error: Unknown keyword or identifier. Did you mean 'from'?
> ```

Thanks for the headsup, let me rectify this :)

**Comment by mark-wiemer** [comment]
@kyklish, see cc7c153

```jsonc
// package.json
    "lint-staged": {
        "*": "prettier --check",
        "src/**/*.ts": [
```

**Comment by mark-wiemer** [comment]
I've also updated the formatting tests in my branch, https://github.com/mark-wiemer/vscode-autohotkey-plus-plus/tree/fix-formatter-bug

**Review by mark-wiemer — APPROVED** [review]
Fantastic work once again :)

**Review by mark-wiemer — DISMISSED** [review]

**Comment by mark-wiemer** [comment]
Can't merge until we're passing the tests. I think the regex rearrangements are responsible here. We should be preserving quoted strings, not trimming them or removing extra spaces.

**Comment by kyklish** [comment]
Yes. Now semicolon inside quoted string became single line comment. I'm thinking about proper regex for comment detection.

**Review by mark-wiemer — APPROVED** [review]

**Pull request merged**